compiler/optimizer/src/main/java/org/apache/nemo/compiler/optimizer/pass/runtime/DynamicTaskSizingRuntimePass.java

```diff
@@ -0,0 +1,124 @@
+import org.apache.nemo.common.ir.edge.executionproperty.SubPartitionSetProperty;
+import org.apache.nemo.common.ir.vertex.IRVertex;
+import org.apache.nemo.common.ir.vertex.executionproperty.EnableDynamicTaskSizingProperty;
+import org.apache.nemo.common.ir.vertex.executionproperty.ParallelismProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Runtime pass for Dynamic Task Sizing policy.
+ */
+public final class DynamicTaskSizingRuntimePass extends RunTimePass<Map<String, Long>> {
+  private static final Logger LOG = LoggerFactory.getLogger(DynamicTaskSizingRuntimePass.class.getName());
+  private final String mapKey = "opt.parallelism";
+
+  /**
+   * Default Constructor.
+   */
+  public DynamicTaskSizingRuntimePass() {
+  }
+
+  @Override
+  public IRDAG apply(final IRDAG irdag, final Message<Map<String, Long>> mapMessage) {
+    final Set<IREdge> edgesToOptimize = mapMessage.getExaminedEdges();
+    final Set<IRVertex> stageVertices = edgesToOptimize.stream().map(IREdge::getDst).collect(Collectors.toSet());
+    irdag.topologicalDo(v -> {
+      if (stageVertices.contains(v)) {
+        edgesToOptimize.addAll(irdag.getIncomingEdgesOf(v));
+      }
+    });
+    LOG.info("Examined edges {}", edgesToOptimize.stream().map(IREdge::getId).collect(Collectors.toList()));
+
+    final IREdge representativeEdge = edgesToOptimize.iterator().next();
+    // double check
+    if (!representativeEdge.getDst().getPropertyValue(EnableDynamicTaskSizingProperty.class).orElse(false)) {
+      return irdag;
+    }
+    final Map<String, Long> messageValue = mapMessage.getMessageValue();
+    LOG.info("messageValue {}", messageValue);
+    final int optimizedTaskSizeRatio = messageValue.get(mapKey).intValue();
+    final int partitionerProperty = getPartitionerProperty(irdag);
+    for (IREdge edge : edgesToOptimize) {
+      if (!edge.getPropertyValue(PartitionerProperty.class).get().right().equals(partitionerProperty)
+      && edge.getPropertyValue(CommunicationPatternProperty.class).get()
+        .equals(CommunicationPatternProperty.Value.SHUFFLE)) {
+        throw new IllegalArgumentException();
+      }
+    }
+    final int partitionUnit = partitionerProperty / optimizedTaskSizeRatio;
+    edgesToOptimize.forEach(irEdge -> setSubPartitionProperty(irEdge, partitionUnit, partitionerProperty));
+    edgesToOptimize.forEach(irEdge -> setDstVertexParallelismProperty(irEdge, partitionUnit, partitionerProperty));
+    return irdag;
+  }
+
+  private int getPartitionerProperty(final IRDAG dag) {
+    long jobSizeInBytes = dag.getInputSize();
+    long jobSizeInGB = jobSizeInBytes / (1024 * 1024 * 1024);
+    if (1 <= jobSizeInGB && jobSizeInGB < 10) {
+      return 1024;
+    } else if (10 <= jobSizeInGB && jobSizeInGB < 100) {
```

Comment:
Names for these numbers would be useful.

e.g., 
static int 2GB = 2048;
static int SMALL_BUCKET(?) = 10;